common/utils/attacks.ts

```diff
@@ -147,77 +147,69 @@ function shouldIgnoreCard(attack: AttackModel, instance: string): boolean {
 export function executeAttacks(
 	game: GameModel,
 	attacks: Array<AttackModel>,
-	withoutBlockingActions = false
+	withoutBlockingActions = false,
+	createLogs = true
 ) {
-	// Outer attack loop
-	while (attacks.length > 0) {
-		const allAttacks: Array<AttackModel> = []
-
-		// Main attack loop
-		while (attacks.length > 0) {
-			// STEP 1 - Call before attack and defence for all attacks
-			runBeforeAttackHooks(attacks)
-			runBeforeDefenceHooks(attacks)
-
-			// STEP 2 - Call on attack and defence for all attacks
-			runOnAttackHooks(attacks)
-			runOnDefenceHooks(attacks)
-
-			// STEP 3 - Execute all attacks
-			for (let i = 0; i < attacks.length; i++) {
-				executeAttack(attacks[i])
-
-				// Add this attack to the final list
-				allAttacks.push(attacks[i])
-			}
-
-			const newAttacks: Array<AttackModel> = []
-			for (let i = 0; i < attacks.length; i++) {
-				newAttacks.push(...attacks[i].nextAttacks)
-				// Clear the list of next attacks on this attack
-				attacks[i].nextAttacks = []
-			}
-			attacks = newAttacks
+	// We need to store the SU card for battle log stuff.
+	const thisAttackSagaSU = game.currentPlayer.board.singleUseCard
+
+	// STEP 1 - Call before attack and defence for all attacks
+	runBeforeAttackHooks(attacks)
+	runBeforeDefenceHooks(attacks)
+
+	// STEP 2 - Call on attack and defence for all attacks
+	runOnAttackHooks(attacks)
+	runOnDefenceHooks(attacks)
+
+	// STEP 3 - Execute all attacks
```

Comment:
These comments don't add clarity. The function names and forEach does all of this for us.
```suggestion
	runBeforeAttackHooks(attacks)
	runBeforeDefenceHooks(attacks)

	runOnAttackHooks(attacks)
	runOnDefenceHooks(attacks)

```

Reply:
Going to keep the comments bc they were like that beforehand. Unless you strongly feel they need to be changed

Reply:
If they were like that before, I think it's fine to leave them

Reply:
I wrote those comments, and for me they add clarity. Generally I wouldn't remove other comments unless they are clearly outdated